src/udev/udev-manager.c

```diff
@@ -149,7 +140,8 @@ Manager* manager_free(Manager *manager) {
         udev_rules_free(manager->rules);
 
         hashmap_free(manager->workers);
-        event_queue_cleanup(manager, EVENT_UNDEF);
+        while (manager->events)
```

Comment:
`LIST_CLEAR(event, manager->events, event_free);`

Reply:
Unfortunately, it does not work, as LIST_CLEAR() removes each entry from the list before calling the free function, but event_free() also tries to remove the entry from the list and triggers assertion...